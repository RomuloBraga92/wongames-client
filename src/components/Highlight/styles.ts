import styled, { css } from 'styled-components'
import media from 'styled-media-query'

import { HighlightProps } from '.'

export type ContainerProps = Pick<
  HighlightProps,
  'backgroundImage' | 'alignment'
>

const containerModifiers = {
  left: () => css`
    grid-template-areas: 'floatimage contentitems';
    grid-template-columns: 1.3fr 2fr;

    ${Content} {
      text-align: right;
    }
  `,

  right: () => css`
    grid-template-areas: 'contentitems floatimage';
    grid-template-columns: 2fr 1.3fr;

    ${Content} {
      text-align: left;
    }
    ${FloatImage} {
      justify-self: end;
    }
  `,
}

export const Container = styled.div<ContainerProps>`
  ${({ backgroundImage, alignment }) => css`
    height: 23rem;
    position: relative;
    background-image: url(${backgroundImage});
    background-position: center center;
    background-size: cover;
    display: grid;

    &::after {
      content: '';
      width: 100%;
      height: 100%;
      position: absolute;
      background: rgba(0, 0, 0, 0.6);
      top: 0;
      left: 0;
    }

    ${media.greaterThan('medium')`
      height: 32rem;
    `}

    ${containerModifiers[alignment!]()}
  `}
`

export const FloatImage = styled.img`
  ${({ theme }) => css`
    z-index: ${theme.layers.base};
    max-height: 23rem;
    max-width: 100%;
    grid-area: floatimage;
    align-self: end;

    ${media.greaterThan('medium')`
      height: 28rem;
    `}
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    z-index: ${theme.layers.base};
    padding: ${theme.spacings.xsmall};
    grid-area: contentitems;

    ${media.greaterThan('medium')`
      align-self: end;
      padding: ${theme.spacings.large}
    `}
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.bold};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xxlarge}
    `}
  `}
`

export const Subtitle = styled.h3`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.light};
    margin-bottom: ${theme.spacings.medium};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.large}
    `}
  `}
`
