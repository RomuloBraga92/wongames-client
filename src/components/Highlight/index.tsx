import * as S from './styles'
import Button from 'components/Button'

export type HighlightProps = {
  title: string
  subtitle: string
  buttonLabel: string
  buttonLink: string
  backgroundImage: string
  floatImage?: string
  alignment?: 'left' | 'right'
}

const Highlight = ({
  title,
  subtitle,
  buttonLabel,
  buttonLink,
  backgroundImage,
  floatImage,
  alignment = 'left',
}: HighlightProps) => (
  <S.Container backgroundImage={backgroundImage} alignment={alignment}>
    {!!floatImage && <S.FloatImage src={floatImage} alt={title} />}
    <S.Content>
      <S.Title>{title}</S.Title>
      <S.Subtitle>{subtitle}</S.Subtitle>
      <Button as="a" href={buttonLink}>
        {buttonLabel}
      </Button>
    </S.Content>
  </S.Container>
)

export default Highlight
