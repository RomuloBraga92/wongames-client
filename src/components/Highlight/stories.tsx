import { Story, Meta } from '@storybook/react/types-6-0'
import Highlight, { HighlightProps } from './'

export default {
  title: 'Highlight',
  component: Highlight,
  args: {
    title: 'Read Dead it’s back',
    subtitle: 'Come see John’s new adventures',
    buttonLabel: 'Buy now',
    buttonLink: '/rdr2',
    backgroundImage: '/img/red-dead-img.jpg',
  },
} as Meta

export const Default: Story<HighlightProps> = args => (
  <div style={{ maxWidth: '104rem' }}>
    <Highlight {...args} />
  </div>
)

export const WithFloatImage: Story<HighlightProps> = args => (
  <div style={{ maxWidth: '104rem' }}>
    <Highlight {...args} />
  </div>
)

Default.parameters = {
  layout: 'fullscreen',
}

WithFloatImage.args = {
  floatImage: '/img/red-dead-float.png',
}

WithFloatImage.parameters = {
  layout: 'fullscreen',
}
