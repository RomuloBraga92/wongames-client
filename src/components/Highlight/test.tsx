import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Highlight from './'

const props = {
  title: 'Heading 1',
  subtitle: 'Heading 2',
  buttonLabel: 'Buy now',
  buttonLink: '/rdr2',
}

describe('<Highlight />', () => {
  it('should render the headings and the button', () => {
    renderWithTheme(<Highlight {...props} />)

    expect(
      screen.getByRole('heading', { name: /Heading 1/i }),
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /Heading 2/i }),
    ).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /buy now/i })).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /buy now/i })).toHaveAttribute(
      'href',
      '/rdr2',
    )
  })
})
