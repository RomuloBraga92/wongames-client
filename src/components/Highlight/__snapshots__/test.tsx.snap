// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render the headings and the button 1`] = `
.c0 {
  height: 23rem;
  position: relative;
  background-image: url(/img/red-dead-img.jpg);
  background-position: center center;
  background-size: cover;
  display: grid;
  grid-template-areas: 'floatimage contentitems';
  grid-template-columns: 1.3fr 2fr;
}

.c0::after {
  content: '';
  width: 100%;
  height: 100%;
  position: absolute;
  background: rgba(0,0,0,0.6);
  top: 0;
  left: 0;
}

.c0 .c1 {
  text-align: right;
}

.c2 {
  z-index: 10;
  padding: 1.6rem;
  grid-area: contentitems;
}

.c3 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 600;
}

.c4 {
  color: #FAFAFA;
  font-size: 1.4rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  border-radius: 0.4rem;
  cursor: pointer;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #FAFAFA;
  border: 0;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 500;
  text-align: center;
  font-size: 1.6rem;
  height: 5rem;
  padding: 0.8rem 4.8rem;
}

.c5:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

@media (min-width:768px) {
  .c0 {
    height: 32rem;
  }
}

@media (min-width:768px) {
  .c2 {
    -webkit-align-self: end;
    -ms-flex-item-align: end;
    align-self: end;
    padding: 4.0rem;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1 c2"
  >
    <h2
      class="c3"
    >
      Heading 1
    </h2>
    <h3
      class="c4"
    >
      Heading 2
    </h3>
    <a
      class="c5"
      href="/rdr2"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</div>
`
